### ISSUE TITLE
PyQgsGeometryTest failed with libgeos 3.9.0

### ISSUE THREAD

**Issue by benoitdm-oslandia** [qgis/QGIS#45516]
### What is the bug or the crash?

From the master, the test PyQgsGeometryTest failed with libgeos 3.9.0 but succeed with the 3.9.1

### Steps to reproduce the issue

1. build master with libgeos 3.9.0
2. run test with `ctest -V -R PyQgsGeometryTest`: 
```
457: ======================================================================
457: FAIL: testReshape (__main__.TestQgsGeometry)
457: Test geometry reshaping
457: ----------------------------------------------------------------------
457: Traceback (most recent call last):
457:   File "/home/bde/prog/2010_25_lille_3d/qgis/tests/src/python/test_qgsgeometry.py", line 2705, in testReshape
457:     self.assertTrue(compareWkt(wkt, expWkt),
457: AssertionError: False is not true : testReshape failed: mismatch Expected:
457: Polygon ((0 0, 1 0, 1 1, 0 1, 0.5 0.5, 0 0))
457: Got:
457: Polygon ((0 0, 1 0, 1 1, 0 1, 0 0))
457: 
457: 
457: ======================================================================
457: FAIL: testTaperedBuffer (__main__.TestQgsGeometry)
457: ----------------------------------------------------------------------
457: Traceback (most recent call last):
457:   File "/home/bde/prog/2010_25_lille_3d/qgis/tests/src/python/test_qgsgeometry.py", line 5673, in testTaperedBuffer
457:     self.assertTrue(compareWkt(result, exp, 0.01),
457: AssertionError: False is not true : tapered buffer: mismatch Expected:
457: MultiPolygon (((2 -0.5, 1.75 -0.43301270189221935, 1.56698729810778081 -0.25000000000000011, 1.5 0.00000000000000006, 1.25 2, 1.35048094716167078 2.37499999999999956, 1.62499999999999978 2.649519052838329, 2 2.75, 2.03076923076923066 2.75384615384615383, 2 3, 2.13397459621556118 3.49999999999999956, 2.5 3.86602540378443837, 3.00000000000000044 4, 3.50000000000000044 3.86602540378443837, 3.86602540378443837 3.5, 4 3, 3.875 1.99999999999999978, 3.75777222831138413 1.56250000000000044, 3.4375 1.24222777168861631, 3 1.125, 2.64615384615384608 1.1692307692307693, 2.5 -0.00000000000000012, 2.43301270189221963 -0.24999999999999983, 2.25 -0.4330127018922193, 2 -0.5)),((2.69433781190019195 3.6142041887027907, 2.51132511000797276 3.79721689059500989, 2.44433781190019195 4.04721689059500989, 2.51132511000797232 4.29721689059500989, 2.69433781190019195 4.48022959248722952, 2.94433781190019195 4.54721689059500989, 5.45950095969289784 5.11976967370441471, 5.95950095969289784 4.98579507748885309, 6.32552636347733621 4.61976967370441471, 6.45950095969289784 4.11976967370441471, 6.3255263634773371 3.61976967370441516, 5.95950095969289784 3.25374426991997634, 5.45950095969289784 3.11976967370441471, 2.94433781190019195 3.54721689059500989, 2.69433781190019195 3.6142041887027907)),((5.5804222648752404 3.94683301343570214, 6.0804222648752404 3.81285841722014052, 6.44644766865967878 3.44683301343570214, 6.5804222648752404 2.94683301343570214, 6.44644766865967966 2.44683301343570259, 6.0804222648752404 2.08080760965126377, 5.5804222648752404 1.94683301343570214, 3 2.5, 2.75 2.56698729810778081, 2.56698729810778081 2.75, 2.5 3, 2.56698729810778037 3.24999999999999956, 2.75 3.43301270189221919, 3 3.5, 5.5804222648752404 3.94683301343570214)))
457: Got:
457: MultiPolygon (((2.64615384615384608 1.1692307692307693, 2.5 -0.00000000000000012, 2.43301270189221963 -0.24999999999999983, 2.25 -0.4330127018922193, 2 -0.5, 1.75 -0.43301270189221935, 1.56698729810778081 -0.25000000000000011, 1.5 0.00000000000000006, 1.25 2, 1.35048094716167078 2.37499999999999956, 1.62499999999999978 2.649519052838329, 2 2.75, 2.03076923076923066 2.75384615384615383, 2 3, 2.13397459621556118 3.49999999999999956, 2.5 3.86602540378443837, 3.00000000000000044 4, 3.50000000000000044 3.86602540378443837, 3.86602540378443837 3.5, 4 3, 3.875 1.99999999999999978, 3.75777222831138413 1.56250000000000044, 3.4375 1.24222777168861631, 3 1.125, 2.64615384615384608 1.1692307692307693)),((2.69433781190019195 3.6142041887027907, 2.51132511000797276 3.79721689059500989, 2.44433781190019195 4.04721689059500989, 2.51132511000797232 4.29721689059500989, 2.69433781190019195 4.48022959248722952, 2.94433781190019195 4.54721689059500989, 5.45950095969289784 5.11976967370441471, 5.95950095969289784 4.98579507748885309, 6.32552636347733621 4.61976967370441471, 6.45950095969289784 4.11976967370441471, 6.3255263634773371 3.61976967370441516, 5.95950095969289784 3.25374426991997634, 5.45950095969289784 3.11976967370441471, 2.94433781190019195 3.54721689059500989, 2.69433781190019195 3.6142041887027907)),((2.75 2.56698729810778081, 2.56698729810778081 2.75, 2.5 3, 2.56698729810778037 3.24999999999999956, 2.75 3.43301270189221919, 3 3.5, 5.5804222648752404 3.94683301343570214, 6.0804222648752404 3.81285841722014052, 6.44644766865967878 3.44683301343570214, 6.5804222648752404 2.94683301343570214, 6.44644766865967966 2.44683301343570259, 6.0804222648752404 2.08080760965126377, 5.5804222648752404 1.94683301343570214, 3 2.5, 2.75 2.56698729810778081)))
457: 
457: 
457: ======================================================================
457: FAIL: testVoronoi (__main__.TestQgsGeometry)
457: ----------------------------------------------------------------------
457: Traceback (most recent call last):
457:   File "/home/bde/prog/2010_25_lille_3d/qgis/tests/src/python/test_qgsgeometry.py", line 5163, in testVoronoi
457:     self.assertTrue(compareWkt(result, exp, 0.00001),
457: AssertionError: False is not true : delaunay: mismatch Expected:
457: GeometryCollection (Polygon ((25 38, 25 295, 221.20588235294118817 210.91176470588234793, 170.02400000000000091 38, 25 38)),Polygon ((400 38, 170.02400000000000091 38, 221.20588235294118817 210.91176470588234793, 400 369.65420560747656964, 400 38)),Polygon ((25 395, 400 395, 400 369.65420560747656964, 221.20588235294118817 210.91176470588234793, 25 295, 25 395)))
457: Got:
457: GeometryCollection (Polygon ((25 295, 221.20588235294118817 210.91176470588234793, 170.02400000000000091 38, 25 38, 25 295)),Polygon ((221.20588235294118817 210.91176470588234793, 400 369.65420560747656964, 400 38, 170.02400000000000091 38, 221.20588235294118817 210.91176470588234793)),Polygon ((25 395, 400 395, 400 369.65420560747656964, 221.20588235294118817 210.91176470588234793, 25 295, 25 395)))
457: 
```

### Versions

* qgis master (1295cb1a62b1e779a56a4ca74a166c59e434b37a)
* libgeos 3.9.0

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by lbartoletti** [comment]
And If I'm not wrong, it works also with GEOS 3.8.0 since this version is used for the CI.

@strk Do you know if it's a regression in 3.9.0 and fixed in 3.9.1?

**Comment by benoitdm-oslandia** [comment]
Too old to be tested.
